tools/cfn2ts/lib/codegen.ts

```diff
@@ -236,10 +236,10 @@ export default class CodeGenerator {
     this.code.openBlock(`public static fromCloudFormation(scope: ${CONSTRUCT_CLASS}, id: string, resourceAttributes: any, options: ${CORE}.FromCloudFormationOptions): ` +
       `${resourceName.className}`);
     this.code.line('resourceAttributes = resourceAttributes || {};');
-    this.code.indent('const cfnParser = new cfn_parse.CfnParser({');
-    this.code.line('finder: options.finder,');
-    this.code.unindent('});');
     if (propsType) {
+      this.code.indent('const cfnParser = new cfn_parse.CfnParser({');
+      this.code.line('finder: options.finder,');
+      this.code.unindent('});');
```

Comment:
How about making `handleAttributes` a method on `CfnParser`? This way, you wouldn't have to change this, and also you wouldn't have to pass `finder` to it (as it already has one).

Thoughts on this idea?